core/lib/dal/src/blocks_dal.rs

```diff
@@ -883,6 +884,118 @@ impl BlocksDal<'_, '_> {
         Ok(())
     }
 
+    pub async fn save_l1_batch_inclusion_data(
+        &mut self,
+        number: L1BatchNumber,
+        da_inclusion_data: Vec<u8>,
+    ) -> anyhow::Result<()> {
+        let update_result = sqlx::query!(
+            r#"
+            UPDATE data_availability
+            SET
+                inclusion_data = $1,
+                updated_at = NOW()
+            WHERE
+                l1_batch_number = $2
+                AND inclusion_data IS NULL
+            "#,
+            da_inclusion_data.as_slice(),
+            i64::from(number.0),
+        )
+        .instrument("save_l1_batch_da_data")
+        .with_arg("number", &number)
+        .report_latency()
+        .execute(self.storage)
+        .await?;
+
+        if update_result.rows_affected() == 0 {
+            tracing::debug!("L1 batch #{number}: DA data wasn't updated as it's already present");
+
+            // Batch was already processed. Verify that existing DA data matches
+            let matched: i64 = sqlx::query!(
```

Comment:
out of curiocity - why don't you fetch the previous value and simply compare? 
(as instead you do a count with a 'complex' comparison)